atest/testdata/running/for.robot

```diff
@@ -317,6 +324,120 @@ For In Range With Expressions Containing Floats
     \    @{result} =    Create List    @{result}    ${i}
     Should Be True    @{result} == [3.14, 1.14, -0.86]
 
+For In Zip
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    : FOR    ${item}    ${thing}    IN ZIP    ${items}    ${things}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}
+    Should Be True    @{result} == ['a:e', 'b:f', 'c:g', 'd:h']
+
+For In Zip With Uneven Lists
+    [Documentation]    Handling lists with different number of elements.  This will ignore any elements after the shortest list ends, just like with Python's zip()
+    @{items}=    Create List    a    b    c
+    @{things}=    Create List    d    e    f    g    h
+    : FOR    ${item}    ${thing}    IN ZIP    ${items}    ${things}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}
+    Should Be True    @{result} == ['a:d', 'b:e', 'c:f']
+
+For In Zip With 3 Lists
+    [Documentation]    Handling more than one list like :FOR ${a} ${b} ${c} IN ZIP ${x} ${y} ${z}.
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    @{stuffs}=    Create List    1    2    3    4    5
+    : FOR    ${item}    ${thing}    ${stuff}    IN ZIP    ${items}    ${things}    ${stuffs}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}:${stuff}
+    Should Be True    @{result} == ['a:e:1', 'b:f:2', 'c:g:3', 'd:h:4']
+
+For In Zip With Other Iterables
+    [Documentation]    Handling non-lists. Should accept anything iterable except strings and fail with a clear error message if invalid data given. You can use utils.is_list_like to verify inputs.
+    @{xrange}=    Evaluate    xrange(10)
+    @{xrange2}=    Evaluate    xrange(10,20)
```

Comment:
- `xrange` is not Python 3 compatible. Generator expression like `(i in range(10)` would be better in that regard.
- Other non-list iterable could just be a tuple.